### PR TITLE
ENH: Adapt fMRIPrep to upcoming sMRIPrep API changes

### PR DESCRIPTION
**Pull request by oesteban**
In particular, poldracklab/fmriprep#167 introduces a few changes that need to be considered before #1829 is merged:

  - The new `skull_strip_mode` argument for the anatomical workflow (poldracklab/smriprep#167)
  - The functional workflow does not require parameterized inputs
  - The anatomical workflow no longer produces a brain-masked conversion of the T1w.

### CODE REVIEW

**Comment by oesteban** [comment]
This build should be ready to pass. Comments are welcome :)

File: setup.cfg [hunk 1]
```diff
@@ -21,16 +21,16 @@ python_requires = >=3.7
...
-    niworkflows @ git+https://github.com/nipreps/niworkflows.git@76f831f1f8fbf6c277fdf92535639bbf11d84527
+    niworkflows ~= 1.1.9
     numpy
     pandas
     psutil >= 5.4
     pybids >= 0.10.0
     pyyaml
     sdcflows ~= 1.2.0
-    smriprep ~= 0.5.2
+    smriprep >= 0.6.0rc1
```
Comment (mgxd) [inline setup.cfg]:
```suggestion
    smriprep ~= 0.6.0rc1
```

We should keep this strict to avoid future API changes (i.e. 0.7.x)

**Comment by oesteban** [comment]
> Looks like it'll only require adding a flag to the CLI and maybe modifying an output, right?

Yup. The option and a new line in the config module for this option.

File: fmriprep/workflows/base.py [hunk 2]
```diff
@@ -222,16 +222,15 @@ def init_single_subject_wf(subject_id):
         freesurfer=config.workflow.run_reconall,
         hires=config.workflow.hires,
         longitudinal=config.workflow.longitudinal,
-        num_t1w=len(subject_data['t1w']),
         omp_nthreads=config.nipype.omp_nthreads,
         output_dir=str(config.execution.output_dir),
         reportlets_dir=reportlets_dir,
         skull_strip_fixed_seed=config.workflow.skull_strip_fixed_seed,
-        # skull_strip_mode='force',
+        skull_strip_mode='force',
```
Comment (mgxd) [inline fmriprep/workflows/base.py]: surely this will have a config option?

File: setup.cfg (hunk 1 above)
Comment (oesteban) [reply setup.cfg]: I might be wrong, but I think `~=` here will not match 0.6.0 (without the rc). What about:
```suggestion
    smriprep >= 0.6.0rc1,<0.7
```

File: fmriprep/workflows/base.py (hunk 2 above)
Comment (oesteban) [reply fmriprep/workflows/base.py]: yes (we are talking about that right now in #1829 )

File: setup.cfg (hunk 1 above)
Comment (mgxd) [reply setup.cfg]: yup, that works.

**Pull request merged**